sklearn/tree/_criterion.pyx

```diff
@@ -1269,6 +1272,9 @@ cdef class MAE(RegressionCriterion):
         cdef SIZE_t i, p, k
         cdef DOUBLE_t y_ik
         cdef DOUBLE_t median
+        cdef DOUBLE_t w = 1.0
+        cdef DOUBLE_t imp_left = 0.0
```

Comment:
can you call `imp_left` to `impurity_left` and `impurity_left` to `p_impurity_left` to differentiate which one is the pointer and like this both name will be explicit.
The same thing should be done for the `right`.

Reply:
Done 👍